plugins/modules/mso_rest.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, Anvitha Jain (@anvitha-jain) <anvjain@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: mso_rest
+short_description: Direct access to the Cisco MSO REST API
+description:
+- Enables the management of the Cisco MSO fabric through direct access to the Cisco MSO REST API.
+- This module is not idempotent and does not report changes.
+options:
+  method:
+    description:
+    - The HTTP method of the request.
+    - Using C(delete) is typically used for deleting objects.
+    - Using C(get) is typically used for querying objects.
+    - Using C(post) is typically used for modifying objects.
+    - Using C(put) is typically used for modifying existing objects.
+    - Using C(patch) is typically also used for modifying existing objects.
+    type: str
+    choices: [ delete, get, post, put, patch ]
+    default: get
+    aliases: [ action ]
+  path:
+    description:
+    - URI being used to execute API calls.
+    type: str
+    required: yes
+    aliases: [ uri ]
+  content:
+    description:
+    - When used instead of C(src), sets the payload of the API request directly.
+    - This may be convenient to template simple requests.
+    - For anything complex use the C(template) lookup plugin (see examples)
+      or the M(template) module with parameter C(src).
+    type: raw
+extends_documentation_fragment:
+- cisco.mso.modules
+
+notes:
+- Certain payloads are known not to be idempotent, so be careful when constructing payloads.
+- For JSON payloads nothing special is needed.
+- If you do not have any attributes, it may be necessary to add the "attributes" key with an empty dictionnary "{}" for value
+  as the MSO does expect the entry to precede any children.
+seealso:
+- module: cisco.mso.mso_tenant
+author:
+- Anvitha Jain (@anvitha-jain)
+'''
+
+EXAMPLES = r'''
+- name: Add tenant
+  cisco.mso.mso_rest:
+    host: mso
+    username: admin
+    password: SomeSecretPassword
+    path: /api/v1/tenants
+    method: post
+    content:
+      {
+          "displayName": "mso_tenant",
+          "name": "mso_tenant",
+          "description": "",
+          "siteAssociations": [],
+          "userAssociations": [],
+          "_updateVersion": 0
+      }
+  delegate_to: localhost
+
+- name: Add schema (check_mode)
```

Comment:
This does not seems to use check_mode so not sure why the title say so. Also it seems to be the same example as the previous one (JSON + POST)